### ISSUE TITLE
Plex Playback issues

### ISSUE THREAD

**Issue by jl94x4** [lostb1t/replex#55]
Okay a few days ago I opened an issue which I can no longer find, relating to playback issues, and after a few tests I've determined is caused by Replex/as a byproduct of Replex. 

When I disable REPLEX my friend has no issues watching files, however with REPLEX enabled, his shows/movies cut out every 5minutes, sometimes not even giving the option to resume and/or skipping to the next episode randomly.

Can you give me any guidance on what may possibly cause this? To the best of my knowledge I have REPLEX set up correctly.

**Comment by jl94x4** [comment]
I did try remote access and via the same url, both worked with no issues with replex disabled.

**Comment by jl94x4** [comment]
He's tried plex web, mobile (android) and also the plex Windows app too.

**Comment by jl94x4** [comment]
> Did you try using redirect streams?

I haven't. I'll be honest, I'm not sure how to use that.

**Comment by jl94x4** [comment]
http://ip:32400

**Comment by jl94x4** [comment]
> Do you by any change use the force_maximum_quality setting? If so I solved an issue with subtitles. Might try the latest version

I'm gonna re-test tomorrow, this could be the issue. I'll report back.

**Comment by jl94x4** [comment]
Getting more and more reports of this happening.

I asked around, and none of them have used subtitles at all. Not sure what the issue is.

**Comment by jl94x4** [comment]
> https://ip.plex.direct:32400

This doesn't load for me either.

**Comment by Metuant** [comment]
Seeing similar behavior on Samsung for Plex client (Tizen)

**Comment by jl94x4** [comment]
Yes, can confirm that removing REPLEX and going back to just remote access fixes the issue.

**Comment by billyors** [comment]
I don't know if it's the same issue being reported here. But I found that on Roku, my playback streams will end early as if they were completed. About 5 minutes into an episode or movie (often times x265/hevc, x264 seems unaffected) the stream will end and act as if it completed the whole title. PMS debug logs aren't helpful here. But I looked at the Roku logs and I can see it marking the stream as "completed" prematurely. Is this the same behavior other clients are seeing?
What's interesting is that browser, HTPC and Android don't seem to have this issue.

I found that I could workaround the Roku issue by disabling bridge networking on plex and using host networking instead. Then replex points to the plex host ip:port and it seems to workaround the issue. Now Roku streams are completing.

**Comment by jl94x4** [comment]
> I don't know if it's the same issue being reported here. But I found that on Roku, my playback streams will end early as if they were completed. About 5 minutes into an episode or movie (often times x265/hevc, x264 seems unaffected) the stream will end and act as if it completed the whole title. PMS debug logs aren't helpful here. But I looked at the Roku logs and I can see it marking the stream as "completed" prematurely. Is this the same behavior other clients are seeing? What's interesting is that browser, HTPC and Android don't seem to have this issue.
> 
> I found that I could workaround the Roku issue by disabling bridge networking on plex and using host networking instead. Then replex points to the plex host ip:port and it seems to workaround the issue. Now Roku streams are completing.

Sounds similar I'm not using Docker for Plex though.

**Comment by Metuant** [comment]
Re-routed - seems to have helped - will read the docs more closely next time, thank you for a great tool!

**Comment by billyors** [comment]
Hoping for the solution here too.
I routed the above paths with traefik and I'm still seeing the issue.

**Comment by Metuant** [comment]
> @Metuant just to be clear
> 
> Routing
> 
> /video/:/transcode/universal/session /library/parts
> 
> Solved the issue for you?

Correct

**Comment by jl94x4** [comment]
Can someone explain how you reroute? The readme isn't that clear and in unsure what to do now.

**Comment by billyors** [comment]
> > Hoping for the solution here too.
> > I routed the above paths with traefik and I'm still seeing the issue.
> 
> Is it in specific case? Like only when transcoding or with subtitles or just always?

Always with hevc files on Roku. Regardless of subtitles on or off its killing the playback after about 5 minutes. I have everything in mkv files so Roku is having to direct stream. The behavior is consistent where it acts as if the file has completed playback and takes you to the next episode or movie selection screen.

**Comment by jl94x4** [comment]
> > > Hoping for the solution here too.
> > > I routed the above paths with traefik and I'm still seeing the issue.
> > 
> > 
> > Is it in specific case? Like only when transcoding or with subtitles or just always?
> 
> Always with hevc files on Roku. Regardless of subtitles on or off its killing the playback after about 5 minutes. I have everything in mkv files so Roku is having to direct stream. The behavior is consistent where it acts as if the file has completed playback and takes you to the next episode or movie selection screen.

Sounds like my case. How do you go about re-routing? I generally have no idea.

**Comment by billyors** [comment]
> > > > Hoping for the solution here too.
> > > > I routed the above paths with traefik and I'm still seeing the issue.
> > > 
> > > 
> > > Is it in specific case? Like only when transcoding or with subtitles or just always?
> > 
> > 
> > Always with hevc files on Roku. Regardless of subtitles on or off its killing the playback after about 5 minutes. I have everything in mkv files so Roku is having to direct stream. The behavior is consistent where it acts as if the file has completed playback and takes you to the next episode or movie selection screen.
> 
> Sounds like my case. How do you go about re-routing? I generally have no idea.

This is entirely dependent on your reverse proxy tool. I'm using traefik which allows you to set labels on docker containers for specific routing rules. I have the following in place. But, this doesn't solve my playback issues.

```
version: '3.8'
services:
  traefik:
    ...
  plex:
    image: plexinc/pms-docker:plexpass
    container_name: plex
    ...
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.plex-rtr.entrypoints=https"
      - "traefik.http.routers.plex-rtr.rule=Host(`replex.custom.domain`) && (PathPrefix(`/video/:/transcode/universal/session`) || PathPrefix(`/library/parts`))"
      - "traefik.http.routers.plex-rtr.tls=true"
      - "traefik.http.services.plex-svc.loadbalancer.server.port=32400"
  replex:
    image: ghcr.io/lostb1t/replex:latest
    container_name: replex
    ...
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.replex-rtr.entrypoints=https"
      - "traefik.http.routers.replex-rtr.rule=Host(`replex.custom.domain`) && !(PathPrefix(`/video/:/transcode/universal/session`) || PathPrefix(`/library/parts`))"
      - "traefik.http.routers.replex-rtr.tls=true"
      - "traefik.http.services.replex-svc.loadbalancer.server.port=80"
```

This essentially says these two paths should be routed directly to plex. Then, that replex should exclude these paths from its routing rules.

**Comment by billyors** [comment]
> Koop> > > Hoping for the solution here too.
> 
> > > > I routed the above paths with traefik and I'm still seeing the issue.
> > 
> > 
> > > 
> > 
> > 
> > > Is it in specific case? Like only when transcoding or with subtitles or just always?
> > 
> > 
> > Always with hevc files on Roku. Regardless of subtitles on or off its killing the playback after about 5 minutes. I have everything in mkv files so Roku is having to direct stream. The behavior is consistent where it acts as if the file has completed playback and takes you to the next episode or movie selection screen.
> 
> Is your plex instance and Replex in the same location? Ea both in the same local network or plex remote etc

Same docker host, same local network. I'm using docker's dns for communication between the two.

**Comment by billyors** [comment]
The bang encompasses both paths `!(path 1 || path 2)`, but maybe if I set them explicitly then it'll make a difference. Testing, will report back.

**Comment by billyors** [comment]
No luck 😒

It's possible my issue is a corner-case involving the hand-off of a media type that doesn't play nice in this double proxy setup between traefik, replex and plex. I've tried a number of things and it narrows down to hevc files (.mkv) struggling to playback with replex on a roku over the LAN. Until someone else can replicate this I can't say for certain this has a high impact for other users.

🤷‍♂️ I mentioned it in a previous post above, but my workaround is just to skip connecting to replex for my roku devices and directly connect to plex.

**Comment by jl94x4** [comment]
> 🤷‍♂️ I mentioned it in a previous post above, but my workaround is just to skip connecting to replex for my roku devices and directly connect to plex.

How are you achieving this?

**Comment by jl94x4** [comment]
> @jl94x4 is it only direct play causing issues for you aswell?

Mine is also items that are transcoded

**Comment by jl94x4** [comment]
Unfortunately, I'm gonna have to stop using REPLEX, more than half my users are reporting this as an issue. Playback stopping after 5/10mins and Plex acting like its been completely watched and then moving onto the next episode. 

This happens whether transcoded, direct played. Seems to be happening mostly on Android, but not limited to just Android.

**Comment by billyors** [comment]
@lostb1t I sent you a private so you can try replicating with the same setup that I'm using if you'd like.


> > 🤷‍♂️ I mentioned it in a previous post above, but my workaround is just to skip connecting to replex for my roku devices and directly connect to plex.
> 
> How are you achieving this?

Roku devices allow you to specify a manual connection under the settings on the client. This is done device-by-device. No magic needed.

**Comment by jl94x4** [comment]
> @lostb1t I sent you a private so you can try replicating with the same setup that I'm using if you'd like.
> 
> > > 🤷‍♂️ I mentioned it in a previous post above, but my workaround is just to skip connecting to replex for my roku devices and directly connect to plex.
> > 
> > 
> > How are you achieving this?
> 
> Roku devices allow you to specify a manual connection under the settings on the client. This is done device-by-device. No magic needed. Although, if I understand _your_ issue then it sounds like playback issues are happening outside of your LAN. In that case, the manual connection won't work very well because you'll need to specify your home IP, which is likely the same thing you're using for the replex redirect (i.e. issue persists).

Yes, this is a remote server. (hosted VPS)

**Comment by jl94x4** [comment]
> Im still unable to replicate
> 
> If you want to help, enable tracing: RUST_LOG="info,replex=trace"
> 
> Then play the media untill it quits.
> 
> Paste the output of Replex to pastebin

Here is a log of exactly the moment that plex just stops playback completely.

https://pastebin.com/whTYx04M

**Comment by jl94x4** [comment]
We're the logs of any help in possibly fixing this @lostb1t ?

**Comment by jl94x4** [comment]
Ahsoka season 1 episode 1 usually 5-7mins in using Windows 11 plex app,
doesn't do it on browser though.

**Comment by jl94x4** [comment]
> @jl94x4 Made a few changes. If you could give it a go

Same exact issue @lostb1t on the same app, tried multiple items seems to be doing it for anything I play.

**Comment by jl94x4** [comment]
Here is a video of exactly what it does.. it is always exactly 5-7mins in regardless of where playback starts. @lostb1t 

https://github.com/lostb1t/replex/assets/18123855/922ac21d-5df3-4fd4-86e4-e2bd80443edb

**Comment by jl94x4** [comment]
Any other progress with this @lostb1t

**Comment by billyors** [comment]
Not lostb1t. But, I've tried numerous things to get this working and the only conclusion I keep coming back to is that some playback devices require special attention. 
I think it's possible our reverse proxies are mishandling the communication from plex client --> reverse proxy --> replex --> plex . Could be the headers, the user permissions, the session ids. There are a lot of variables. But why only select devices? 🧠💥

**Comment by jl94x4** [comment]
> Not lostb1t. But, I've tried numerous things to get this working and the only conclusion I keep coming back to is that some playback devices require special attention. I think it's possible our reverse proxies are mishandling the communication from plex client --> reverse proxy --> replex --> plex . Could be the headers, the user permissions, the session ids. There are a lot of variables. But why only select devices? 🧠💥

Are you having the same behaviour as in my [video above](https://github.com/lostb1t/replex/issues/55#issuecomment-1713903449)

**Comment by billyors** [comment]
> > Not lostb1t. But, I've tried numerous things to get this working and the only conclusion I keep coming back to is that some playback devices require special attention. I think it's possible our reverse proxies are mishandling the communication from plex client --> reverse proxy --> replex --> plex . Could be the headers, the user permissions, the session ids. There are a lot of variables. But why only select devices? 🧠💥
> 
> Are you having the same behaviour as in my [video above](https://github.com/lostb1t/replex/issues/55#issuecomment-1713903449)

Yes, same issue.

**Comment by billyors** [comment]
I think I finally got this working 🎉. But I don't know what changed. I re-routed these paths in traefik:
```
/video/:/transcode
/library/parts
```

I had tested this before with lostb1t and we didn't have any luck. I've made several changes in traefik (plus the recent versions of replex) so it's hard to tell what might've caused the fix. I kept working at this until I removed all of the 400 status errors from the replex log. There appears to be one more for `music/:/transcode/` that I'm planning to re-route as well.

**Comment by jl94x4** [comment]
> I think I finally got this working 🎉. But I don't know what changed. I re-routed these paths in traefik:
> 
> ```
> /video/:/transcode
> /library/parts
> ```
> 
> I had tested this before with lostb1t and we didn't have any luck. I've made several changes in traefik (plus the recent versions of replex) so it's hard to tell what might've caused the fix. I kept working at this until I removed all of the 400 status errors from the replex log. There appears to be one more for `music/:/transcode/` that I'm planning to re-route as well.

Interesting. I will try routing those two, though I'm using Nginx Proxy Manager, I have no experience with traefik or caddy and its not something I want to really get into, as my entire setup would need changing from the bottom up, which seems like a big learning curve and pretty timeconsuming.

**Comment by jl94x4** [comment]
Update:

I re-routed the `/video/:/transcode` specifically and this seems to have worked. Before I was routing `/video/:/transcode/universal/session`

So I do feel like the `/video/:transcode/ is the one that has fixed my issue. I did try it again earlier today (before routing this) and it was still not working. 

This time I'm over 8 minutes in on the video playing and it hasn't stopped. Usually it'd have gone off by the 5th minute every single time!

Definitely going to be re-routing the `/music/:/transcode` too.

**Comment by billyors** [comment]
Looks like it still works great with allowing the `video/:/transcode/universal/decision` path through replex.

**Comment by billyors** [comment]
> Looks like it still works great when routing the `video/:/transcode/universal/decision` path through replex.

I spoke too soon. I ran through my playback tests and on the second title I got the same issue as before. Playback completing/ending too early. It's possible I misconfigured traefik. Having a rule for the root `/video/:/transcode/` for plex and a subpath for replex at `video/:/transcode/universal/decision` it's possible traffic is getting intercepted unintentionally.

**Comment by jl94x4** [comment]
Interesting I will keep trying other titles to see if I get the same issue again.

**Comment by billyors** [comment]
> Interesting I will keep trying other titles to see if I get the same issue again.

Just routing the root `/video/:/transcode/` seems to fix the issue. The issues I'm seeing now are when trying to route the subpath to get the other features of replex working, like the force maximum quality.

**Comment by jl94x4** [comment]
> > Interesting I will keep trying other titles to see if I get the same issue again.
> 
> Just routing the root `/video/:/transcode/` seems to fix the issue. The issues I'm seeing now are when trying to route the subpath to get the other features of replex working, like the force maximum quality.

Interesting. I don't use the force maximum quality, or other features like that so I can't really help with that part. 

I've noticed over the last couple of months, since Plex upped the default playback from 4MB to 12MB most people on my server direct play much more than they used to, so I don't really need to use force maximum quality!

**Comment by billyors** [comment]
@lostb1t My diagnosis of this weirdness. I _think_ what's happening is that the `decision` path generates a session id for plex auth. When replex calls this path and gets the 200 OK, it then sends the magic onto plex. But, when plex receives the parameters for playback, it is not recognizing the session id as valid.

`Denying access due to session lacking permission to transcode key /library/metadata`

That is why, _I think_™, by routing the entire `/video/:/transcode/` path this isn't an issue. The session id generated in the decision path is valid when routed through solely through plex.

I could be way off, but this stands valid from the testing I've been doing (at least in regards to Roku). Let me know if I can test further.

**Comment by billyors** [comment]
> Where do you see this error? In the plex log?

Correct. PMS log.

**Comment by billyors** [comment]
To be clear, the WARN message does contain the id number as well. `/library/metadata/1234567` not sure why I left that bit off.

Correct, I can only recreate this error when routing the decision path through replex. When the entire `/video/:/transcode/` path is routed through plex the issue is resolved.

This error/warning is the only consistent log I have. Everything else from replex, traefik and plex looks healthy.

**Comment by jl94x4** [comment]
Is there anything else to route @billyors because I'm still getting reports of things going back to the play menu and having to resume the movie. Same app as before.

**Comment by jl94x4** [comment]
This seems to be getting worse by the day @lostb1t any ideas what else we can try? I'd rather not stop using REPLEX if possible

**Comment by jl94x4** [comment]
> > This seems to be getting worse by the day @lostb1t any ideas what else we can try? I'd rather not stop using REPLEX if possible
> 
> Did you try rerouting everything under: /video/:/transcode

/library/parts
/video/:/transcode
/video/:/transcode/universal/decision

I have those rerouted, is there anything I've missed?

**Comment by jl94x4** [comment]
@lostb1t here are my logs for my server. A few points to note;

1. Chromecast with Google TV sometimes takes 2-3 attempts at playing an item (it constantly spins, have to back out and try to start the show/movie again) It usually works after 2-3 goes, this does not happen when REPLEX is disabled. Seems to be the only device that has this issue. Roku is perfectly fine.
2. The playback stopping on Plex for Windows 11 is still an issue, and maybe you can find something regarding that in these logs.

**Comment by jl94x4** [comment]
docker run --rm -it -p 90:80 -e REPLEX_HOST="http://ip:32400" -e REPLEX_AUTO_SELECT_VERSION=true -e REPLEX_DISABLE_RELATED=true -e REPLEX_INCLUDE_WATCHED=true  -e REPLEX_HERO_ROWS=tv.toprated,tv.inprogress,hub.movie.recentlyreleased -e REPLEX_CACHE_ROWS=true -e REPLEX_CACHE_ROWS_REFRESH=true -e REPLEX_CACHE_TTL=0 ghcr.io/lostb1t/replex:latest

I do not have any replex logs, no.

**Comment by eandrade92630** [comment]
Just came to say I now have 2 people in my family telling me this is happening to them. But I have not been able to get them to help me capture logs when it happens.

**Comment by eandrade92630** [comment]
I'm curious. What version of Android do your devices have that have this issue? And it's it Android tv or mobile?

**Comment by billyors** [comment]
It's been a little bit since I've come back to this. My success rate is like 50/50 on this issue. I think it might be the reverse proxy having an issue with some clients. Here's what I've consistently found. 

This is my error (curious to know if other proxy systems show a similar error):
`reverseproxy.go:667: httputil: ReverseProxy read error during body copy: unexpected EOF`

Without a doubt, this log indicates that plex playback through replex (proxied by traefik) will 100% of the time fail and act as if the title has completed. Debug indicates an issue with the path `/video/:/transcode/universal/start.mkv` like below:
```
level=debug msg="Request has been aborted [192.168.87.110:33423 - /video/:/transcode/universal/start.mkv?protocol=http&mediaIndex=0&waitForSegments=1&X-Plex-Client-Profile-Extra=append-transcode-target-codec%28type%3DvideoProfile%26context%3Dstreaming%26protocol%3Dhttp%26videoCodec%3Dvp9%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dvp9%26type%3DupperBound%26name%3Dvideo.bitDepth%26value%3D10%26isRequired%3Dfalse%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dhevc%26type%3DnotMatch%26name%3Dvideo.DOVIProfile%26value%3D5%29%2Badd-limitation%28scope%3DvideoTranscodeTarget%26scopeName%3Dhevc%26scopeType%3DvideoCodec%26context%3Dstreaming%26protocol%3Dhttp%26type%3Dmatch%26name%3Dvideo.colorTrc%26list%3Dbt709%7Cbt470m%7Cbt470bg%7Csmpte170m%7Csmpte240m%7Cbt2020-10%7Cbt2020-10%7Csmpte2084%26isRequired%3Dfalse%29%2Bappend-transcode-target-codec%28type%3DvideoProfile%26context%3Dstreaming%26protocol%3Dhttp%26videoCodec%3Dhevc%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dhevc%26type%3DupperBound%26name%3Dvideo.bitDepth%26value%3D10%26isRequired%3Dfalse%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dh264%26type%3DupperBound%26name%3Dvideo.level%26value%3D41%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dh264%26type%3DupperBound%26name%3Dvideo.refFrames%26value%3D8%26replace%3Dtrue%29%2Badd-transcode-target%28type%3DsubtitleProfile%26context%3Dstreaming%26protocol%3Dhttp%26container%3Dwebvtt%26subtitleCodec%3Dwebvtt%29%2Badd-limitation%28scope%3DvideoAudioCodec%26scopeName%3Dmp3%26type%3DupperBound%26name%3Daudio.channels%26replace%3Dtrue%26value%3D0%29%2Bappend-transcode-target-audio-codec%28type%3DvideoProfile%26context%3Dstreaming%26protocol%3Dhttp%26audioCodec%3Deac3%29%2Badd-limitation%28scope%3DvideoAudioCodec%26scopeName%3Deac3%26type%3DupperBound%26name%3Daudio.channels%26value%3D8%26isRequired%3Dfalse%29&videoQuality=100&session=9f5ac382-e299-457c-8b35-0d1524435d8e-490&X-Plex-Token=Redacted&mediaBufferSize=50000&offset=1158&partIndex=0&videoResolution=3840x2160&directPlay=0&copyts=1&path=%2Flibrary%2Fmetadata%2F518170&directStream=1&subtitles=auto]: net/http: abort Handler" middlewareName=traefik-internal-recovery middlewareType=Recovery
```
My guess is that the behavior differs client to client because of the way the transcode session is handled. At least on Roku it uses `/video/:/transcode/universal/start` and for the browser it uses `/video/:/transcode/universal/session/`. I never see any issues on browser.

Some of the replex code in `src/routes.rs` handles these paths. But, this behavior happens still when I'm redirecting the `/video/:/transcode/` path directly to plex as opposed to letting replex handle it. 

The reason I say this issue is 50/50 is because if I restart plex, replex, and traefik it's hit or miss whether or not this behavior happens again. Thoughts?

**Comment by jl94x4** [comment]
> It's been a little bit since I've come back to this. My success rate is like 50/50 on this issue. I think it might be the reverse proxy having an issue with some clients. Here's what I've consistently found.
> 
> This is my error (curious to know if other proxy systems show a similar error): `reverseproxy.go:667: httputil: ReverseProxy read error during body copy: unexpected EOF`
> 
> Without a doubt, this log indicates that plex playback through replex (proxied by traefik) will 100% of the time fail and act as if the title has completed. Debug indicates an issue with the path `/video/:/transcode/universal/start.mkv` like below:
> 
> ```
> level=debug msg="Request has been aborted [192.168.87.110:33423 - /video/:/transcode/universal/start.mkv?protocol=http&mediaIndex=0&waitForSegments=1&X-Plex-Client-Profile-Extra=append-transcode-target-codec%28type%3DvideoProfile%26context%3Dstreaming%26protocol%3Dhttp%26videoCodec%3Dvp9%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dvp9%26type%3DupperBound%26name%3Dvideo.bitDepth%26value%3D10%26isRequired%3Dfalse%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dhevc%26type%3DnotMatch%26name%3Dvideo.DOVIProfile%26value%3D5%29%2Badd-limitation%28scope%3DvideoTranscodeTarget%26scopeName%3Dhevc%26scopeType%3DvideoCodec%26context%3Dstreaming%26protocol%3Dhttp%26type%3Dmatch%26name%3Dvideo.colorTrc%26list%3Dbt709%7Cbt470m%7Cbt470bg%7Csmpte170m%7Csmpte240m%7Cbt2020-10%7Cbt2020-10%7Csmpte2084%26isRequired%3Dfalse%29%2Bappend-transcode-target-codec%28type%3DvideoProfile%26context%3Dstreaming%26protocol%3Dhttp%26videoCodec%3Dhevc%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dhevc%26type%3DupperBound%26name%3Dvideo.bitDepth%26value%3D10%26isRequired%3Dfalse%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dh264%26type%3DupperBound%26name%3Dvideo.level%26value%3D41%29%2Badd-limitation%28scope%3DvideoCodec%26scopeName%3Dh264%26type%3DupperBound%26name%3Dvideo.refFrames%26value%3D8%26replace%3Dtrue%29%2Badd-transcode-target%28type%3DsubtitleProfile%26context%3Dstreaming%26protocol%3Dhttp%26container%3Dwebvtt%26subtitleCodec%3Dwebvtt%29%2Badd-limitation%28scope%3DvideoAudioCodec%26scopeName%3Dmp3%26type%3DupperBound%26name%3Daudio.channels%26replace%3Dtrue%26value%3D0%29%2Bappend-transcode-target-audio-codec%28type%3DvideoProfile%26context%3Dstreaming%26protocol%3Dhttp%26audioCodec%3Deac3%29%2Badd-limitation%28scope%3DvideoAudioCodec%26scopeName%3Deac3%26type%3DupperBound%26name%3Daudio.channels%26value%3D8%26isRequired%3Dfalse%29&videoQuality=100&session=9f5ac382-e299-457c-8b35-0d1524435d8e-490&X-Plex-Token=Redacted&mediaBufferSize=50000&offset=1158&partIndex=0&videoResolution=3840x2160&directPlay=0&copyts=1&path=%2Flibrary%2Fmetadata%2F518170&directStream=1&subtitles=auto]: net/http: abort Handler" middlewareName=traefik-internal-recovery middlewareType=Recovery
> ```
> 
> My guess is that the behavior differs client to client because of the way the transcode session is handled. At least on Roku it uses `/video/:/transcode/universal/start` and for the browser it uses `/video/:/transcode/universal/session/`. I never see any issues on browser.
> 
> Some of the replex code in `src/routes.rs` handles these paths. But, this behavior happens still when I'm redirecting the `/video/:/transcode/` path directly to plex as opposed to letting replex handle it.
> 
> The reason I say this issue is 50/50 is because if I restart plex, replex, and traefik it's hit or miss whether or not this behavior happens again. Thoughts?

Does this also happen for you when direct playing both video and audio files? Because it does for me.

I've never had this issue on Roku, or Chromecast but I have experienced buffering that I don't experience on any other device on the chromecast, which doesn't happen when not using replex.

I also don't have this issue on browser.

**Comment by billyors** [comment]
I haven't confirmed if this happens with direct play. I don't know how to recreate this issue on demand and can't verify at the moment. I would think this doesn't happen with direct play though. That seems strange. I'll update if I start seeing this issue again.

**Comment by eandrade92630** [comment]
I'm assuming you guys are using a reverse proxy in front of replex, like nginx or caddy or something? I'm curious, are you guys passing through any headers?

**Comment by eandrade92630** [comment]
What sort of video are you guys playing when you get this issue? Is it a 4K video? Or less than 4K?

**Comment by billyors** [comment]
Went down the rabbit hole with this issue again. I think I found a silver-ish bullet to the issue. This line here in the below repository is what helped resolve _some_ of my playback woes: 
https://github.com/toomuchio/plex-nginx-reverseproxy/blob/master/nginx.conf#L19

This GHI covers the root cause:
https://github.com/toomuchio/plex-nginx-reverseproxy/issues/21

tldr; set your reverse proxy timeout for transmitting a response to the client to ~`100m`. Nginx does this with `send_timeout 100m`. I think traefik (my proxy) requires a few settings but I think this did it for me:
```
  [http.serversTransports]
    idleConnTimeout = "100m"
    responseHeaderTimeout = "100m"
```

Again, this has helped solve _some_ issues. I'm still testing the timeout value. I have some files that still end abruptly even with this set.

**Comment by jl94x4** [comment]
Its unfortunate @billyors that this hasn't been fixed. I'm getting more and more reports of this on multiple devices (xbox being the latest) and its concerning that the developer of this app hasn't been active for 3 weeks, so I'm at the point that I will likely move away from REPLEX even though I do love the layout of it, but unfortunately for me, practicality has to come first, if people can't watch things, what's the point in it looking nice? 

I may revisit replex in the future if its development is picked up, and these sort of issues fixed. I'll definitely be keeping an eye out on the project @lostb1t in the event that it returns stronger!

**Comment by jl94x4** [comment]
I'll keep an eye on this thread too @billyors incase you find a permanent fix!

**Comment by eandrade92630** [comment]
@jl94x4 you don't have to give up on the project indefinitely. There are features that are still nice such as replex hero rows and such. Features that don't require access to the transcoding endpoint. If you don't require anything that needs access to the transcoding end point just reroute them directly to Plex.

**Comment by jl94x4** [comment]
> @jl94x4 you don't have to give up on the project indefinitely. There are features that are still nice such as replex hero rows and such. Features that don't require access to the transcoding endpoint. If you don't require anything that needs access to the transcoding end point just reroute them directly to Plex.

I would, but I'm not sure how to reroute them directly to Plex. This is something I would definitely like to do as I love the layout that REPLEX creates.

**Comment by jl94x4** [comment]
Is it just a case of enabling REPLEX_REDIRECT_STREAMS?

**Comment by jl94x4** [comment]
> > > @jl94x4 you don't have to give up on the project indefinitely. There are features that are still nice such as replex hero rows and such. Features that don't require access to the transcoding endpoint. If you don't require anything that needs access to the transcoding end point just reroute them directly to Plex.
> > 
> > 
> > I would, but I'm not sure how to reroute them directly to Plex. This is something I would definitely like to do as I love the layout that REPLEX creates.
> > Is there an easy way for you to help me figure this out?
> 
> The method is listed on the github page AND was talked about earlier in this thread.
> 
> `If you have a reverse proxy running and dont want to proxy streaming through plex then you can route the following paths and it subpaths directly to plex.
> 
> ```
> /video/:/transcode/universal/session
> /library/parts
> ```
> 
> `
> 
> So... read the docs for whichever reverse proxy you're using and learn how to route paths.
> 
> This isn't an issue with replex, I don't think, is more an issue of you needing to learn how your reverse proxy solution works.

I've already re routed those paths and still get the same issue as the other people I'm this thread. I said I would stop using reflex before someone said I can still use the hero styles whilst not having replex handling streaming. That's where the confusion has arisen. I've routed all the paths without success (with the issue this thread is about)

**Comment by eandrade92630** [comment]
@jl94x4 can you open a new thread in discussions stating the http server you're using and I'll try to help you get it sorted out.

**Comment by jl94x4** [comment]
> @jl94x4 can you open a new thread in discussions stating the http server you're using and I'll try to help you get it sorted out.

Sure! Thanks

**Comment by jl94x4** [comment]
https://github.com/lostb1t/replex/issues/55#issuecomment-1751284578

Any update on this @billyors

**Comment by jl94x4** [comment]
I am not having any issues with streams ending abruptly now. Tested extensively.

**Comment by lostb1t** [comment]
thats great to hear. will close this. if anybody experiences any issues it can be reopened
